schema_based_on_diagram.sql

```diff
@@ -0,0 +1,58 @@
+--create patients table
+CREATE TABLE patients(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	name CHAR(255),
+	date_of_birth DATE
+);
+
+-- create medical_hitories
+CREATE TABLE medical_histories(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	admitted_at TIME,
+	patient_id INT,
+	status CHAR(255)
+);
+
+-- create invoices table
+
+CREATE TABLE invoices(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	total_amount DECIMAL,
+	generated_at TIME,
+	payed_at TIME,
+	medical_history_id INT
+);
+
+--create invoice_items
+
+CREATE TABLE invoice_items(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	unit_price DECIMAL,
+	quantity INT,
+	total_price DECIMAL,
+	invoice_id INT,
+	treatment_id INT
+);
+
+-- create treatments table
+CREATE TABLE treatments(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	type CHAR(255),
+	name CHAR(255)
+);
+
+-- add table relationships
+-- for one-to-many relations
+
+ALTER TABLE medical_histories ADD FOREIGN KEY (patient_id) REFERENCES patients(id);
+ALTER TABLE invoices ADD FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id);
+ALTER TABLE invoice_items ADD FOREIGN KEY (invoice_id) REFERENCES invoices(id);
+ALTER TABLE invoice_items ADD FOREIGN KEY (treatment_id) REFERENCES treatments(id);
```

Comment:
- [OPTIONAL] Altering tables is nice. But it would have been nicer to just add the relationships on the fly while creating your tables. This comes with one caveat though, you have to order your schema in such a way that a table must have been created before trying to refer to it when creating a relationship.

  This could have been another option:
  
  ```sql
  -- create medical_hitories
  CREATE TABLE medical_histories(
	  id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
	  admitted_at TIME,
	  patient_id INT REFERENCES patients(id),
	  status CHAR(255)
  );
  
  -- create invoices table  
  CREATE TABLE invoices(
	  id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
	  total_amount DECIMAL,
	  generated_at TIME,
	  payed_at TIME,
	  medical_history_id INT REFERENCES medical_histories(id)
  );
  
  --create invoice_items  
  CREATE TABLE invoice_items(
	  id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
	  unit_price DECIMAL,
	  quantity INT,
	  total_price DECIMAL,
	  invoice_id INT REFERENCES invoices(id),
	  treatment_id INT REFERENCES treatments(id)
  );
  ```
  
  All in all, well done for setting the Foreign Keys 👍🏼.

Reply:
Thanks for the amazing feedback!
I will proceed to make the necessary changes :100: